pkg/conference/participant/tracker.go

```diff
@@ -75,6 +76,16 @@ func (t *Tracker) RemoveParticipant(participantID ID) map[string]bool {
 		}
 	}
 
+	// Remove this participant's subscriptions.
+	for _, subscribers := range t.subscribers {
+		for subscriberID, subscription := range subscribers {
+			if subscriberID == participantID {
+				subscription.Unsubscribe()
+				delete(subscribers, subscriberID)
+			}
+		}
+	}
```

Comment:
```suggestion
	for _, subscribers := range t.subscribers {
		for subscriberID, subscriber := range subscribers {
			if subscriberID == participantID {
				subscriber.Unsubscribe()
				delete(subscribers, subscriberID)
			}
		}
	}
```

Reply:
The thing is that the subscriber is **a key** in the map. subscription is **a value**. The actual type that is inside the value is a `Subscription`, that's why I also called the variable `subsscription`.

Perhaps the confusion comes from the fact that `subscribers` is **a map of maps**. `subscribers` is a map where the key is a track ID, while the value is a map that holds the correspondence between the subscriber ID (participant ID) and the subscription, i.e. contains both subscribers and subscriptions.

Or, if it would be simple to understand, `subscription` is semantically a map `map[TrackID]SubscriptionDetails`, where `SubscriptionDetails` is:

```go
type SubscriptionDetails struct {
    subscriber participant.ID
    subscription subscription.Subscription
}
```

Does it make any sense? 🙂

Reply:
In other words: a subscriber is a participant (or rather, its identifier) who is subscribed to something. Subscription is an actual entity (the actual subscription that handles the logic for the subscribed track, processes incoming packets and writes them to the output track).

Reply:
Ah, sorry I am confused